packages/apps/dashboard/ui-2024/src/features/Leaderboard/components/DataGrid.tsx

```diff
@@ -0,0 +1,131 @@
+import { LeaderBoardData } from '@services/api/use-leaderboard-details';
+import { Box, Typography } from '@mui/material';
+import { handleErrorMessage } from '@services/handle-error-message';
+import Loader from '@components/Loader';
+import { useBreakPoints } from '@utils/hooks/use-is-mobile';
+import { DataGrid as MuiDataGrid } from '@mui/x-data-grid';
+import { useDataGrid } from '../hooks/useDataGrid';
+
+export const DataGrid = ({
+	data = [],
+	status,
+	error,
+}: {
+	data: LeaderBoardData | undefined;
+	status: 'success' | 'error' | 'pending';
+	error: unknown;
+}) => {
+	const { columns, visibleRows } = useDataGrid(data);
+	const {
+		mobile: { isMobile },
+	} = useBreakPoints();
+
+	const tableIsEmpty = status === 'success' && visibleRows.length === 0;
+	const tableMinHeight = status === 'success' && !tableIsEmpty ? 'unset' : 300;
+
+	return (
+		<Box width="100%" height={tableMinHeight} sx={{ overflow: 'hidden' }}>
+			<MuiDataGrid
+				disableColumnResize
+				hideFooterPagination
+				disableColumnMenu
+				disableColumnSelector
+				disableRowSelectionOnClick
+				hideFooter
+				loading={status === 'pending'}
+				rows={visibleRows}
+				columns={columns}
+				getRowId={(row) => row.address}
+				autosizeOptions={{
+					expand: true,
+				}}
+				onSortModelChange={(data) => {
+					console.log('dskodosk', data);
```

Comment:
I assume that file `sorting.ts` was removed because because here new implementation will come

Reply:
yes, datagrid is supporting it out of the box